src/vision/status_quo/aws_engineer/missed_waker_leads_to_lost_performance.md

```diff
@@ -0,0 +1,12 @@
+# Status quo of an AWS engineer: Missed `Waker` leads to lost performance
+
+Once the server is working, Alan starts to benchmark it. He is not really sure what to expect, but he is hoping to see an improvement in performance relative to the baseline service they were using before. To his surprise, it seems to be running slower!
+
+Digging into the problem a bit, he wishes -- not for the first time -- that he had better tools to understand what was happening. This time he doesn't even bother trying to fire up the debugger. He just starts looking at his metrics. He's accumulated a pretty decent set of metrics by now, and he can often get a picture of the state of the runtime from them.
```

Comment:
Apologies if this should be a PR instead of a comment. Let me know and I can perhaps submit one tonight.

It's not clear how does Alan just start looking at his metrics. Here's how I imagine the story goes:

After trying a few common tricks to improve performance without avail, Alan decides to add more metrics and logs in his service, to understand where the bottlenecks are. Alan is used to using a well-supported internal tool (or a mature open source project) to collect metrics, where all he needed to do was pull in the library and set up a few configuration parameters.

However, in Rust, there is no widely-used, battle-tested library inside and outside his company. Even less so in an async code base! So Alan just used what seemed to be the best options: `tracing` and `metrics` crate, but he quickly found that they couldn't do a few of the things he wants to do, and somehow invoking the metrics is causing his service to be even slower. Now, Alan has to debug and profile his metrics implementation before he can even fix his service. (Cue another story on how that's difficult...)

Reply:
Thanks!

Reply:
added